### PR TITLE
Deprecate secio to noise and fix docker build.

### LINKED ISSUE #1214: Deprecate secio
Related: https://github.com/libp2p/go-libp2p/pull/972

Cluster nodes should be backwards compatible because we are using TLS preferentially to secio, so removing it should be no problem. We should add noise as first option though (`clusterhost.go`).

### PR DESCRIPTION
**Pull request by Jorropo**
Closes #1214

### CODE REVIEW

**Comment by Jorropo** [comment]
@hsanjuan this should also fix the docker build (it was calling an non existent image).

File: clusterhost.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func baseOpts(psk corepnet.PSK) []libp2p.Option {
 		libp2p.PrivateNetwork(psk),
 		libp2p.EnableNATService(),
 		libp2p.Security(libp2ptls.ID, libp2ptls.New),
-		libp2p.Security(secio.ID, secio.New),
+		libp2p.Security(noise.ID, noise.New),
```
Comment (hsanjuan) [inline clusterhost.go:121]: Noise should be first (move before tls).

**Review by hsanjuan — CHANGES_REQUESTED** [review]
Small ask. Thanks for contributing this!

**Comment by Jorropo** [comment]
@hsanjuan done

**Comment by hsanjuan** [comment]
@Jorropo it seems docker is still borked

**Comment by Jorropo** [comment]
@hsanjuan it works now (on my machine at least) :)

**Comment by Jorropo** [comment]
It finally builds !

**Review by hsanjuan — APPROVED** [review]

**Pull request merged**